diff --git a/ios/SecureStoreModule.swift b/ios/SecureStoreModule.swift
index 584df218365e2ebf176f6d332970a3ba64d7a12e..d0cd59ae0f2fa69666a0a0f381aefaeec2d8c520 100644
--- a/ios/SecureStoreModule.swift
+++ b/ios/SecureStoreModule.swift
@@ -188,6 +188,12 @@ public final class SecureStoreModule: Module {
       query[kSecAttrAccessGroup as String] = accessGroup
     }
 
+    #if os(tvOS)
+    if #available(tvOS 16.0, *), options.useUserIndependentKeychain {
+      query[kSecUseUserIndependentKeychain as String] = true
+    }
+    #endif
+
     return query
   }
 
diff --git a/ios/SecureStoreOptions.swift b/ios/SecureStoreOptions.swift
index 7e3fa4dfca55cdb9fada4b7fb794b8d91ef0e15c..f9d709f2b7ebea8cf5dbdf565f59c05ddeb2fce1 100644
--- a/ios/SecureStoreOptions.swift
+++ b/ios/SecureStoreOptions.swift
@@ -1,6 +1,9 @@
 import ExpoModulesCore
 
 internal struct SecureStoreOptions: Record {
+  @Field
+  var useUserIndependentKeychain: Bool = false
+
   @Field
   var authenticationPrompt: String?
 
